RevenueCatUI/Templates/Components/PaywallComponentViewModel.swift

```diff
@@ -29,6 +28,7 @@ enum PaywallComponentViewModel {
 extension PaywallComponent {
 
     func toViewModel(
+        packageValidator: PackageValidator,
```

Comment:
Hmm instead of passing the validator around the view models, would it be possible to perform an initial validation before starting to create views/viewmodels? I think it would be cleaner if after we can just assume it's valid. (There might be a reason why that's difficult not really sure, so FFTI).

Reply:
@tonidero This is what I _wanted_ to do however, it would required traversing the component tree twice (since there is no standard placement for packages)

So I _figured_ if we were already traversing/remapping the components to view models, might as well pull out the information we need at the same time?

It feels gross from a code aspect but felt better in performance? 🤷‍♂️

Reply:
Maybe we can abstract this away somehow. What about a wrapper function, e.g. `buildInitialState()`, that wraps `toViewModel()` and calls `isValid` at the end, and returns the root VIewModel + `PaywallState`?

Reply:
> Maybe we can abstract this away somehow. What about a wrapper function, e.g. `buildInitialState()`, that wraps `toViewModel()` and calls `isValid` at the end, and returns the root VIewModel + `PaywallState`?

I dooooo like an idea of this! I think I'd like to come back to that a little later though 🤷‍♂️  Once we have things out and have a little more time to breath? Or until this absolutely feels horrible? 😛

Reply:
Yea it's not that much in the way, for sure! I'm okay with keeping it around for now.